ocaml/xapi/cpuid_helpers.ml

```diff
@@ -121,33 +121,10 @@ let get_flags_for_vm ~__context vm cpu_info =
   let onboot = Map_check.getf ~default:migration features_field_boot cpu_info in
   (vendor, migration, onboot)
 
-(** Upgrade a VM's feature set based on the host's one, if needed.
- *  The output will be a feature set that is the same length as the host's
- *  set, with a prefix equal to the VM's set, and extended where needed.
- *  If the current VM set has 4 words, then we assume it was last running on
- *  a host that did not support "feature levelling v2". In that case, we cannot
- *  be certain about which host features it was using, so we'll extend the set
- *  with all current host features. Otherwise we'll zero-extend. *)
+(** Upgrade a VM's feature set based on the host's one by zero-extending, if needed. *)
 let upgrade_features ~__context ~vm host_features vm_features =
-  let len = Array.length vm_features in
   let upgraded_features =
-    if len <= 4 then
```

Comment:
Should we refuse to zero extend if the feature is too old? i.e. raise an exception if `len <= 4` with an error saying that the VM is unsupported because it has been created pre 7.x.
That might help find such VMs easier (rather than a more subtle failure due to featuresets being wrong.

Reply:
OK, good idea.

Reply:
I don't think this has been addressed yet

Reply:
I think this should be addressed

Reply:
Done now.